lattice-control/lattice-control-interface.smithy

```diff
@@ -37,7 +37,7 @@ service LatticeController {
                  GetHostInventory, GetClaims, ScaleActor,
                  StartActor, AdvertiseLink, RemoveLink,
                  GetLinks, UpdateActor, StartProvider,
-                 StopProvider, StopActor, StopHost]
+                 StopProvider, StopActor, StopHost, SetRegistryCredentials]
```

Comment:
While we're adding this, do we also want to go ahead and configure `DelRegistryCredentials`?

Reply:
This is a deliberate wipe. If a host had credentials prior to this, it should dump them.

Reply:
You could theoretically affect a delete/dump by publishing a set with an empty map.

Reply:
Ok, then this should be fine

Reply:
I struggle with wanting to add/remove individual registries. My gut reaction is that this can create split-brain where one host has one version of a registry and the other host has a different one (this message is broadcast to all hosts). This message is the entirety of all known registries. The set of registries hosts should use for authenticated artifact pull should (imho) be changed all at once - atomic replace.

Reply:
> I still think we should have a way to remove a registry, pushing empty creds is fine as long as we agree that we should check for that and remove it on the config service side

This message wouldn't push just one set of empty creds, if we send this message empty it will delete all credentials.

Reply:
> This message contains all known valid registry credential sets. This message isn't just one registry, it's all of them, designed to be an atomic total replace of all credentials.

I didn't fully get this part before, makes much more sense to have a single `Set` operation